dashboard/app/controllers/teacher_dashboard_controller.rb

```diff
@@ -1,8 +1,10 @@
 class TeacherDashboardController < ApplicationController
-  load_and_authorize_resource :section
+  load_and_authorize_resource :section, :course
 
   def show
-    @section = @section.summarize
+    @section_summary = @section.summarize
+    @section = @section
+    @course = @course
```

Comment:
these two lines shouldn't be necessary. `@section` as defined by `load_and_authorize_resource` should still be accessible in the view file without it